models/src/Enemies.cpp

```diff
@@ -1,30 +1,67 @@
 #include "../include/Enemies.hpp"
-
+#include <RNG.hpp>
+const int SAFE_ZONE = 2;
+const int TOTAL_ENEMIES = 3; 
 Enemies::Enemies()
 {
-	populate(5);
+	// populate();
 }
 
 Enemies::~Enemies()
 {
 	for(auto &e:this->list)
 		delete e;
 }
-void Enemies::updateAll(float deltaTime, const Map &map)
+void Enemies::updateAll(float deltaTime, const Map &map, const Player &player)
 {
 	for(auto &e:this->list)
 	{
-		e->update(deltaTime, map);
+		e->update(deltaTime, map, player);
 	}
 }
 
 void Enemies::populate()
 {
-	//Used for testing new Enemies
-	IEnemy *e =new IEnemy();
-	Ballom *b = new Ballom();
-	list.push_back(e);
-	list.push_back(b);
+	Finder *f = new Finder();
+	list.push_back(f);
+}
+
+void Enemies::populate(int numEnemies,int level, const Map &map){
+	int enemyType;
+	IEnemy *e;
+	//generate coordinates here
+
+	//only generates on uneven blocks
+	//check if tile at coordinates exist
+	//rerun random number generator
+	for(int i = 0; i < numEnemies;i++)
+	{
+		sf::Vector2f start(0,0);
+		while(map.tileAt(sf::Vector2i(start)) != Tile::Clear)
+		{
+			int x = RNG::getRandomNumber(SAFE_ZONE,((map.size().x/2)-1)) *2+1;
+			int y = RNG::getRandomNumber(SAFE_ZONE,((map.size().y/2)-1)) *2+1;
+			start = sf::Vector2f(x+0.5, y+0.5);
+		}
+
+		enemyType = RNG::getRandomNumber(0,level);
+		switch (enemyType)
```

Comment:
It seems like an enum for enemy types would be useful here. If that's not possible, feel free to resolve this without comment.

Reply:
Might try to implement this in future but since I have no idea how we plan on initialising Enemies yet, I'd rather have it working here first, where it can be easily changed without having to jump back and forth between 10 different files.